sys/fido2/ctap/ctap_crypto.c

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright (C) 2021 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup fido2_ctap
+ * @{
+ * @file
+ *
+ * @author      Nils Ollrogge <nils-ollrogge@outlook.de>
+ * @}
+ */
+
+#define ENABLE_DEBUG    (0)
+#include <string.h>
+
+#include "debug.h"
+#include "assert.h"
+#include "random.h"
+
+#include "crypto/ciphers.h"
+#include "crypto/modes/ccm.h"
+#include "crypto/modes/cbc.h"
+
+#include "uECC.h"
+#include "tiny-asn1.h"
+
+#include "fido2/ctap/ctap_crypto.h"
+#include "fido2/ctap/ctap_status.h"
+#include "fido2/ctap/ctap_utils.h"
+
+static ctap_crypto_key_agreement_key_t g_ag_key;
+static int sig_to_der_format(uint8_t *r, uint8_t *s, uint8_t *sig,
+                             size_t *sig_len);
+static int RNG(uint8_t *dest, size_t size);
+
+int fido2_ctap_crypto_init(void)
+{
+
+    uECC_set_rng(&RNG);
```

Comment:
uECC_set_rng expects a function with signature `int (*uECC_RNG_Function)(uint8_t *dest, unsigned size);`. `void fido2_ctap_crypto_prng(uint8_t *buf, size_t size)` returns void.

Reply:
Then why `static int RNG(uint8_t *dest, size_t size)` in line 39?
(using `size_t` instead of `unsigned int` as in the copied function signature?)

However, I am fine with the additional abstraction and was just wondering about the amount of indirection without any changing prototypes.